## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 513978`
Commit: [`20f79496247d721dc1dffbe3a4a278093195947f`](https://github.com/NixOS/nixpkgs/commit/20f79496247d721dc1dffbe3a4a278093195947f) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/20f79496247d721dc1dffbe3a4a278093195947f..pull/513978/head))
Merge: [`f0d46f4308bd326963f1b5bfd8d3cdfc4c2d3d1a`](https://github.com/NixOS/nixpkgs/commit/f0d46f4308bd326963f1b5bfd8d3cdfc4c2d3d1a)

Logs: https://github.com/amadejkastelic/nixpkgs-review-gha/actions/runs/24991617299


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>idescriptor</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>idescriptor</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>idescriptor</summary>
<pre>     |                      ---------------------------------- arguments to this function are incorrect
 100 |             conn.handle(),
 101 |             b&quot;main\0&quot;.as_ptr() as *const i8,
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*const u8`, found `*const i8`
     |
     = note: expected raw pointer `*const u8`
                found raw pointer `*const i8`
note: function defined here
    --&gt; /build/source/build/cargo/rust_2a2b5/aarch64-unknown-linux-gnu/release/build/libsqlite3-sys-c773549f7dc001fe/out/bindgen.rs:2644:12
     |
2644 |     pub fn sqlite3_deserialize(
     |            ^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0308`.
warning: idescriptor_rust_codebase@0.1.0: Could not open /nix/store/sp1xlmzf32zi635v27qlbpdf2793dzlj-qtbase-6.11.0/lib/libQt6QuickControls2.prl file to read libraries to link: No such file or directory (os error 2)
warning: idescriptor_rust_codebase@0.1.0: Could not open /nix/store/sp1xlmzf32zi635v27qlbpdf2793dzlj-qtbase-6.11.0/lib/libQt6Qml.prl file to read libraries to link: No such file or directory (os error 2)
error: could not compile `idescriptor_rust_codebase` (lib) due to 1 previous error
make[2]: *** [CMakeFiles/_cargo-build_idescriptor_rust_codebase.dir/build.make:70: CMakeFiles/_cargo-build_idescriptor_rust_codebase] Error 101
make[1]: *** [CMakeFiles/Makefile2:292: CMakeFiles/_cargo-build_idescriptor_rust_codebase.dir/all] Error 2
make: *** [Makefile:156: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/id/idescriptor/package.nix
+++ b/pkgs/by-name/id/idescriptor/package.nix
@@ -2,11 +2,14 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch,
   buildGoModule,
   nix-update-script,
   copyDesktopItems,
   makeDesktopItem,
+  cargo,
   cmake,
+  corrosion,
   pkg-config,
   avahi-compat,
   ffmpeg,
@@ -25,17 +28,33 @@
   pugixml,
   qt6,
   lxqt,
+  rustPlatform,
+  rustc,
 }:
 stdenv.mkDerivation (finalAttrs: {
   pname = "idescriptor";
-  version = "0.3.0";
+  version = "0.5.0";
 
   src = fetchFromGitHub {
     owner = "iDescriptor";
     repo = "iDescriptor";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-tBTAJqXDqWAqrxQlEEi2kDcVqrB6WrBquxvKV2dkpQ4=";
+    hash = "sha256-AN3CVR9WWa9cG6C6q+hiDyTomT+RebHC1ghr6XyEtAo=";
     fetchSubmodules = true;
+  };
+
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/iDescriptor/iDescriptor/commit/fc73e3146dc4884cf9bc1f7879574ac832cc21e6.patch";
+      hash = "sha256-WqEpSY/fhbsMv0bgU2Ak5japUdohaN7zsNG1BbxJnKs=";
+    })
+  ];
+
+  cargoRoot = "src/rust";
+
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit (finalAttrs) src cargoRoot;
+    hash = "sha256-PJhMb+lMiu8ubOYVX8YVkQzeQMbBO+i6NQhvuyrCujk=";
   };
 
   ipatool-go-modules =
@@ -50,15 +69,19 @@ stdenv.mkDerivation (finalAttrs: {
     }).goModules;
 
   nativeBuildInputs = [
+    cargo
     cmake
     copyDesktopItems
     pkg-config
     go
     qt6.wrapQtAppsHook
+    rustPlatform.cargoSetupHook
+    rustc
   ];
 
   buildInputs = [
     avahi-compat
+    corrosion
     ffmpeg
     libheif
     libimobiledevice
@@ -83,9 +106,17 @@ stdenv.mkDerivation (finalAttrs: {
     lxqt.qtermwidget
   ];
 
+  cxx-qt-cmake = fetchFromGitHub {
+    owner = "kdab";
+    repo = "cxx-qt-cmake";
+    tag = "0.8.1";
+    hash = "sha256-kXSIU71iHn+SSGikGoNeMbBpSrDJ6hwhnHslmskm8nY=";
+  };
+
   cmakeFlags = [
     "-DPACKAGE_MANAGER_MANAGED=ON"
     "-DPACKAGE_MANAGER_HINT=nixpkgs"
+    "-DFETCHCONTENT_SOURCE_DIR_CXXQT=${finalAttrs.cxx-qt-cmake}"
   ];
 
   preConfigure = ''

